### ISSUE TITLE
Received message data is not returned as complete

### ISSUE THREAD

**Issue by etsdriver** [CodeSpartan/UE4TcpSocketPlugin#22]
Hi!

I setup a basic connection with reference image and sent some data from other working application to Unreal. Wireshark sees that data was sent correctly as following (hex):
00 01 00 00 00 06 01 03 04 E8 00 01 83

So excepted return Message is 13 items with following values:
0,1,0,0,0,6,1,3,4,232,0,1,131

But for some reason Unreal side seem to only get 9 items into the message:
![image](https://github.com/CodeSpartan/UE4TcpSocketPlugin/assets/11080471/8ff4ed55-65de-4559-bdc1-7b1b097eb887)

Is the ReadFromInbox() function "skipping" something when receiving data or something else happening here?

**Comment by CodeSpartan** [comment]
No, of course there's no skipping. I'm not sure what the issue is in your example. The only guess I could venture is that you're looking at a TCP message in its entirety in Wireshark, which has overhead (headers), while Unreal only reads the message itself. But off the top of my head, the headers come before the message, while in your example it's the beginning of the message that comes through correctly.

**Comment by etsdriver** [comment]
After good sleep I noticed that the answer I was getting was a real Modbus exception response, where machine flips the upmost bit up on data no 7(adds 128) and responds with errorcode 2 (data 8).

The request i made to PLC was wrong, but when i was looking at it on wireshark, it was fine because the communication was other server i had polling the same data that i was excepting.  

**TLDR;
Problem was between table and chair.**
